src/components/AuthForm/ForgotPasswordPage.jsx

```diff
@@ -0,0 +1,235 @@
+
+	const onVerifyEmail = async event => {
+		event.preventDefault();
+
+		try {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: true }));
+			const { status } = await verifyEmail(email);
+			console.log("isEmailFound :", status);
+
+			if (status === 200) {
+				setForgotPasswordState(prev => ({
+					...prev,
+					isEmailVerified: true
+				}));
+				showToast("success", "An OTP sent successfully to your Email..");
+			} else {
+				showToast("error", "Could'nt find your email..");
+			}
+		} catch (error) {
+			console.error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const onVerifyOTP = async event => {
+		event.preventDefault();
+
+		if (!otp || otp.length < 6) {
+			showToast("error", "Please Enter Valid Otp!");
+			return;
+		}
+
+		try {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: true }));
+
+			const response = await verifyOTP(email, otp);
+
+			if (response.status === 200) {
+				showToast("success", "OTP verified successfully!");
+				setForgotPasswordState(prev => ({
+					...prev,
+					isOTPVerified: true,
+					verificationToken: response.data.token
+				}));
+			} else if (response.code === 401) {
+				showToast("error", "Invalid OTP..");
+			}
+		} catch (error) {
+			throw new Error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const onResendOtp = async () => {
+		try {
+			const { status } = await verifyEmail(email);
+
+			if (status === 200) {
+				setForgotPasswordState(prev => ({
+					...prev,
+					isEmailVerified: true
+				}));
+			} else {
+				showToast("error", "Could'nt find your email..");
+			}
+		} catch (error) {
+			throw new Error(error);
+		}
+	};
+
+	const onNewPasswordSubmit = async event => {
+		event.preventDefault();
+
+		if (passwordError || confirmPasswordError) {
+			showToast("error", "Invalid Password");
+			return;
+		} else if (password !== confirmPassword) {
+			showToast("error", "Password and confirm Password must match..");
+			return;
+		}
+
+		try {
+			setLoaderText("Updating password");
+			setForgotPasswordState(prev => ({
+				...prev,
+				isVerifying: true
+			}));
+			const { status } = await setNewPassword(email, password, verificationToken);
+			if (status === 200) {
+				showToast("success", "Password Changed Successfully..");
+				navigate("/auth");
+			}
+		} catch (error) {
+			throw new Error(error);
+		} finally {
+			setForgotPasswordState(prev => ({ ...prev, isVerifying: false }));
+		}
+	};
+
+	const loader = (
+		<div className="loader-wrapper">
+			<h1 className="text">{loaderText}</h1>{" "}
+			<ReactLoading
+				type="balls"
+				color="#fff"
+				className="balls-loader"
+			/>
+		</div>
+	);
+
+	return (
+		<div id="forgot-password-page-container">
+			<VerifyEmail
+				emailValue={email}
+				onChange={handleInputChange}
+				emailError={emailError}
+				onEmailSubmit={onVerifyEmail}
+			/>
```

Comment:
@Ayush-Bharadva 



`VerifyEmail` is not well encapsulated. I mean all parent component/page needs to know is can is that the if email is verified so it can move user to next step show user OTP screen. let's move the code/state to handle email input, error handling of email and verify email call inside `VerifyEmail`  component.

With the new structure 👆 I am suggesting here verify email will just need one prop:

```
<VerifyEmail 
// handleNext will be pass by parent. 
 onVerify={handleNext}
/>

handleNext(Step) => {
 steps = step + 1
}
```
so instead of `isEmailVerified`, `isOTPVerified`  we will have: 
```
Step === ResetPassword.verifyEmail && <VerifyEmail>

and likewise for others

````

Note: ResetPassword will be a constant like this:

```
ResetPassword: {
 verifyEmail: 1
 verifyOTP: 2
 setNewPassword: 3
}

```

Now why is the suggested structure better? because it boosts reusability. Let's say we want to implement this functionality in userprofile as well. Now if you have everything(code/state) in `forgetPassword.jsx`  you will either have to copy some of the same code or pass a prop make the whole thing conditional like you are doing in your `AddActivityForm`